components/base-rest-api/packages/services/lib/authentication-service.js

```diff
@@ -94,6 +94,8 @@ class AuthenticationService extends Service {
         { token, issuer: claims.iss },
         providerConfig,
       );
+      const tokenSwapperService = await this.service('tokenSwapperService');
+      await tokenSwapperService.swap({ token, uid });
```

Comment:
This needs a documentation update as it adds the requirement that all authentication providers provide a token revocation service. I find this to be a likely case, but it should be noted as it is a new requirement. (tokenSwapperService will throw and authentication will otherwise fail)

Reply:
I agree that we should document this. This is already requirement with sign-out needing a token revocation service.